### PR TITLE
rgw: fix compile errors with boost 1.70

### PR DESCRIPTION
**Pull request by cbodley**
fixes the rgw compilation errors from api changes in boost::asio v1.70

### CODE REVIEW

**Comment by cbodley** [comment]
oof. i was hoping to make jenkins test the builds for me, but they are building `-DWITH_SYSTEM_BOOST=ON`

**Comment by cbodley** [comment]
i removed the cmake changes that raised the boost requirement to 1.70. the rgw commits are unchanged from the ones that built successfully against 1.70

@adamemerson can you please help with review?

**Comment by adamemerson** [comment]
@cbodley We have our own boost repository that Jenkins installs so we'll have to have someone build boost and push to that before we can bump CMake, unfortunately

File: src/rgw/rgw_reshard.h [hunk 1]
```diff
@@ -183,7 +183,14 @@ class RGWReshardWait {
   ceph::condition_variable cond;
 
   struct Waiter : boost::intrusive::list_base_hook<> {
-    boost::asio::basic_waitable_timer<Clock> timer;
+#if BOOST_VERSION < 107000
```
Comment (adamemerson) [inline src/rgw/rgw_reshard.h]: Maybe put a `#warning` here so once we bump the version we can remove the conditional.

**Comment by wjwithagen** [comment]
@cbodley 
Would this PR be eligible for backporting to Nautilus 
And perhaps even further, but I have not gotten to Mimic yet.

**Comment by cbodley** [comment]
absolutely @wjwithagen

**Comment by wjwithagen** [comment]
@cbodley 
https://tracker.ceph.com/issues/42247

**Pull request merged**